sale_loyalty_auto_refresh/models/sale_coupon_refresh_mixin.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2021 Tecnativa - David Vidal
+# Copyright 2021 Camptocamp - Silvio Gregorini
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleCouponRefreshMixin(models.AbstractModel):
+    _name = "sale.coupon.refresh.mixin"
+    _description = "Mixin class for sale coupon auto-refresh features"
+
+    auto_refresh_coupon_triggers_data = fields.Binary(
+        attachment=False,
+        compute="_compute_auto_refresh_coupon_triggers_data",
+        store=False,
+    )
+
+    @api.model
+    def _get_auto_refresh_coupons_triggers(self) -> set:
+        """Returns set of fields which trigger the recomputation.
+
+        Configure additional triggers via config parameters, setting them up
+        as list of fields separated by commas. For example, for sale.order,
+        the param key would be sale_loyalty_auto_refresh.sale_order_triggers,
+        and the value something like: "warehouse_id,carrier_id".
+
+        The method is overriden in the proper modules to set some basic triggers though
+        """
+        additional_triggers = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("sale_loyalty_auto_refresh.%s_triggers" % (self._table), "")
+            .replace(" ", "")
+            .split(",")
+        )
+        return {x for x in additional_triggers if x in self._fields}
+
+    @api.depends(lambda self: list(self._get_auto_refresh_coupons_triggers()))
```

Comment:
Are we sure this complexity w/ the check on new triggers is necessary?
Data:
* the field you compute is non stored -> will get recomputed at every request
* any change on the config param won't have any effect on this depends
* in fact what you are doing is to load the triggers at need (eg: create/write)

Wouldn't be better to have an empty `api.depends`?